src/models.ts

```diff
@@ -27,21 +44,29 @@ export type PokemonType = {
   };
 };
 
-export const getPokemonTypeDamage = async (url: string): Promise<PokemonTypeDamage> => {
+const pokemonTypeDamageCache = new Map<string, PokemonTypeDamage>();
+
+export const fetchPokemonTypeDamage = async (
+  url: string
+): Promise<PokemonTypeDamage> => {
+  if (pokemonTypeDamageCache.has(url)) {
+    return pokemonTypeDamageCache.get(url)!;
+  }
   const response = await fetch(url);
-  const data = await response.json();
-  return data;
+  const pokemonTypeDamage = await response.json();
+  pokemonTypeDamageCache.set(url, pokemonTypeDamage);
+  return pokemonTypeDamage;
 };
 
-export interface PokemonTypeDamage {
+export type PokemonTypeDamage = {
```

Comment:
really they are the same ... but my **personal** recommendation is to use `type` instead of `interface` 
there is a lot of posts about this ts-war. Really you can choose any side, anyway

https://dev.to/manuartero/my-take-on-interface-vs-type-hc8

> remember, it's called TypeScript, not InterfaceScript